Update docs on terminating groups of tasks (#731)
tokio-rs/website: content/tokio/topics/shutdown.md

## Waiting for things to finish shutting down

Once you have told other tasks to shut down, you will need to wait for them to
finish. One easy way to do so is to use a [task tracker]. A task tracker is a

> collection of futures. It is itself a future which resolves only after all of

Reviewer: To emphasize that the futures are spawned, use the word "task".
```suggestion
collection of tasks. It is itself a future which resolves only after all of
```

> collection of futures. It is itself a future which resolves only after all of

Reviewer: No, it is not a future. Instead, it has a method that returns once the tasks are done.

## Waiting for things to finish shutting down

Once you have told other tasks to shut down, you will need to wait for them to
finish. One easy way to do so is to use a [task tracker]. A task tracker is a

> collection of tasks. The [wait] method of the task tracker gives you a future

Reviewer: Method names should be marked as code. (You also have to update the link.)
```suggestion
collection of tasks. The [`wait`] method of the task tracker gives you a future
```

> which resolves only after all of its contained futures have resolved **and** it
> has been closed.

Reviewer: Instead of using "it", let's use "task tracker" to reduce the chance for confusion.
```suggestion
which resolves only after all of its contained futures have resolved **and** the
task tracker has been closed.
```
